### PR TITLE
Unpin transformers for v5.0.0

### PR DESCRIPTION
**Pull request by jveitchmichaelis**
Checked if upstream fixes allow training DETR or if the CI is still breaking. Cause seems to be garbage weight initialisation by transformers, which caused extreme gradients that failed from-scratch training (nan boxes in output). This would then fail the low level tests in the CI, but shouldn't affect using pre-trained weights.

This seems to be fixed and released now, and the CI is pulling 5.0.0. This only affected a narrow use-case and should not matter to most users, so if someone hasn't updated, it shouldn't break other things.

I don't think it's necessary to restrict `transformers` versions in `pyproject` as everything else should work fine with 4.57, but we could include that (e.g. `transformers<4.57, transformers>=5.0.0`)

### CODE REVIEW

**Review by ethanwhite — APPROVED** [review]
LGTM!

@henrykironde - unless you see a reason not to merge this then go ahead and push the big green button
